projects/packages/password-checker/tests/php/test-class-password-checker.php

```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * Tests the Password_Checker package.
+ *
+ * @package automattic/jetpack-password-checker
+ */
+
+namespace Automattic\Jetpack;
+
+use WorDBless\BaseTestCase;
+
+/**
+ * Test Password_Checker class
+ */
+class Password_Checker_Test extends BaseTestCase {
+
+	/**
+	 * Test the password checker.
+	 */
+	public function test_password() {
+		$password_checker = new Password_Checker( null );
+
+		$test_results = $password_checker->test( '123', true );
+		$this->assertFalse( $test_results['passed'] );
+
+		$test_results = $password_checker->test( 'password', true );
+		$this->assertTrue( $test_results['passed'] );
```

Comment:
This seems like it could really use more tests...

Reply:
Indeed!
I will be adding more tests as I learn how.

Reply:
Some Tests that we should add:

- Validate each individual Test [ I'd recommend that we use the filter `password_checker_tests` to remove all but the rule under test. This allows testing of individual rules(tests) and bypass the immediate failure rules. ]
 This is where a data provider could come in handy pass $rule, $password, $expected_result and you can then unhook the other rules, then run `test` passing the password and verifying the result matches the expected defined in data provider.
  - no_backslashes
  - minimum_length
  - has_mixed_case
  - has_digit
  - has_special_char
  - not_a_common_password
  - not_same_as_other_user_data
- Verify required and fail_immediately properties are adhered too.
- Validate filters
  - password_checker_common_passwords. [ add a password to the list and validate the test fails ]
  - password_checker_minimum_password_length [ increase/decrease the length and validate it pass/fails appropriately ]
  - password_checker_minimum_entropy_bits [ increase/decrease the limit and verify passwords pass/fail ]
  - password_checker_tests [ add new test and verify it is run ]

Reply:
Made some progress on this.
I "think" I'm doing it right... but I could be wrong :)
The dataProvider thing for phpunit is super cool!

Reply:
Looks like you're doing it right!

One tip, when your data provider returns a normal array the test names in the ouput will be along the lines of `TestClass::test_method with data set #1`. You can return an associative array instead to get output like `TestClass::test_method with data set "some descriptive key"`, which can make it easier to figure out which one is failing.

Reply:
I was also thinking of passing the `$output_message` arg to the `assertSame` as a 3rd parameter that way it gets output in the results.

Reply:
I've updated the test so that the individual rules are now tested with each entry in the dataProvider.
This takes care of the `preg_match` section of the password tests.
I'm not sure how to go about making tests for the `compare_to_list` section though.
Same with the `fail_immediately` property.